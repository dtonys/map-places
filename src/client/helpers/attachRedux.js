import { Component } from 'react';
import PropTypes from 'prop-types';
import { wrapDisplayName } from 'recompose';
import { Provider as ReduxStoreProvider } from 'react-redux';

import createStore from 'redux-modules/createStore';
import makeAction, {
  execute,
} from 'helpers/reduxAction';

import {
  ACTION_INCREMENT_COUNTER,
} from 'redux-modules/actions/counter';

let clientStore = null;
let serverStore = null;
let serverStoreInitialState = null;
function AttachReduxWithArgs(/* args */) {

  function AttachRedux( WrappedComponent ) {

    class AttachReduxHOC extends Component {
      static propTypes = {
        serverStoreInitialState: PropTypes.object,
      }

      static async getInitialProps( nextJSContext ) {

        if ( __SERVER__ ) {
          const { res } = nextJSContext;
          serverStore = createStore();
          serverStore.dispatch( makeAction(
            execute(ACTION_INCREMENT_COUNTER)
          ) );
          serverStoreInitialState = serverStore.getState();
          // Pass the store to the child components, for SSR.
          res.locals.reduxStore = serverStore;
        }
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(nextJSContext);
        }
        return {
          ...wrappedComponentInitialProps,
          serverStoreInitialState,
        };
      }

      constructor( props ) {
        super(props);
        if ( __CLIENT__ ) {
          if ( !clientStore ) {
            clientStore = createStore( props.serverStoreInitialState );
            window.store = clientStore;
          }
        }
      }

      render() {
        const store = serverStore || clientStore;

        return (
          <ReduxStoreProvider store={store} >
            <WrappedComponent {...this.props} />
          </ReduxStoreProvider>
        );
      }
    }

    AttachReduxHOC.displayName = wrapDisplayName( WrappedComponent, 'AttachReduxHOC' );
    return AttachReduxHOC;
  }
  return AttachRedux;
}
export default AttachReduxWithArgs;
