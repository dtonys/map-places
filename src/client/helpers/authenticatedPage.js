import { Component } from 'react';
import { wrapDisplayName } from 'recompose';
import lodashDifference from 'lodash/difference';

import { Router } from 'routes/pageRoutes';

import {
  extractCurrentUser,
} from 'redux-modules/reducers/user';

// @authenticated()
// @authenticated({ requiredRoles: ['admin', 'member'] })
function AuthenticatedPageWithOptions( options = {} ) {

  function AuthenticatedPage( WrappedComponent ) {

    class AuthenticatedPageHOC extends Component {
      static authRedirect( context, authenticated ) {
        if ( __SERVER__ ) {
          const { req, res } = context;
          const redirectPath = authenticated ? '/' : `/login?next=${encodeURIComponent(req.originalUrl)}`;
          res.redirect(redirectPath);
        }
        if ( __CLIENT__ ) {
          const redirectPath = authenticated ? '/' : `/login?next=${encodeURIComponent(context.asPath)}`;
          Router.replaceRoute(redirectPath);
        }
      }

      static async getInitialProps( context ) {
        const { store } = context;
        const globalState = store.getState();

        const currentUser = extractCurrentUser(globalState);
        if ( !currentUser ) {
          AuthenticatedPageHOC.authRedirect(context, false);
          return {};
        }
        if (
          options.requiredRoles &&
          lodashDifference(options.requiredRoles, currentUser.roles).length !== 0
        ) {
          AuthenticatedPageHOC.authRedirect(context, Boolean(currentUser));
          return {};
        }
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(context);
        }
        return wrappedComponentInitialProps;
      }

      render() {
        return <WrappedComponent {...this.props} />;
      }

    }
    AuthenticatedPageHOC.displayName = wrapDisplayName( WrappedComponent, 'AuthenticatedPageHOC' );
    return AuthenticatedPageHOC;
  }
  return AuthenticatedPage;
}

export default AuthenticatedPageWithOptions;
