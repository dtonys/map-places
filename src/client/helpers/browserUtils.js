export function tryGetCurrentPosition() {
  return new Promise((resolve, reject) => {
    // HTML5 geolocation
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition((position) => {
        resolve({
          lat: position.coords.latitude,
          lng: position.coords.longitude,
        });
      }, () => {
        console.log('geolocation.getCurrentPosition failed.'); // eslint-disable-line no-console
        reject();
      });
    }
    else {
      console.log('Browser doesn\'t support Geolocation.'); // eslint-disable-line no-console
      reject();
    }
  });
}

export function placeHolder() { }
