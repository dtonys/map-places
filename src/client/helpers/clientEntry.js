// HOC for client side entry
import { Component } from 'react';
import { wrapDisplayName } from 'recompose';

import MapManager from 'helpers/MapManager';


let clientEntryRun = false;
// NOTE: Perform client side initialization here.  This is run once only.
function clientEntry() {
  MapManager.initialize({
    initialCenterLatLng: {
      lat: 37.549854,
      lng: -122.291353,
    },
  });
}


function ClientEntryWithArgs(/* args */) {

  function ClientEntry( WrappedComponent ) {

    class ClientEntryHOC extends Component {
      static async getInitialProps( nextJSContext ) {
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(nextJSContext);
        }
        return wrappedComponentInitialProps;
      }

      constructor(props) {
        super(props);
        // Run client entry, once
        if ( __CLIENT__ ) {
          if ( !clientEntryRun ) {
            clientEntry();
            clientEntryRun = true;
          }
        }
      }

      render() {
        return <WrappedComponent {...this.props} />;
      }
    }

    ClientEntryHOC.displayName = wrapDisplayName( WrappedComponent, 'ClientEntryHOC' );
    return ClientEntryHOC;
  }
  return ClientEntry;
}

export default ClientEntryWithArgs;
