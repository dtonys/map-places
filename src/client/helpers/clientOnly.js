import { Component } from 'react';
import { wrapDisplayName } from 'recompose';


function clientOnly( WrappedComponent ) {

  class clientOnlyHOC extends Component {

    constructor( props ) {
      super(props);
      this.state = {
        clientLoaded: false,
      };
    }

    componentDidMount() {
      this.setState({ // eslint-disable-line
        clientLoaded: true,
      });
    }

    render() {
      const { clientLoaded } = this.state;
      if ( clientLoaded ) {
        return ( <WrappedComponent {...this.props} /> );
      }
      return (<div></div>);
    }
  }
  clientOnlyHOC.displayName = wrapDisplayName( WrappedComponent, 'clientOnlyHOC' );
  return clientOnlyHOC;
}

export default clientOnly;
