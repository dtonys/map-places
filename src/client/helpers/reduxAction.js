export const REQUESTED_SUFFIX = '_REQUESTED';
export const STARTED_SUFFIX = '_STARTED';
export const SUCCESS_SUFFIX = '_SUCCESS';
export const ERROR_SUFFIX = '_ERROR';
export const EXECUTED_SUFFIX = '_EXECUTED';

export default function makeAction( type, payload, meta = {} ) {
  return {
    type,
    payload,
    meta,
  };
}

// Send a request to the saga.  Only one may run at a time.
export function request( type ) {
  return `${type}${REQUESTED_SUFFIX}/request`;
}

// Send a request to the saga.  Many can run in parallel.
export function requestFork( type ) {
  return `${type}${REQUESTED_SUFFIX}/requestFork`;
}

// Report an API start
export function apiStart( type ) {
  return `${type}${STARTED_SUFFIX}/apiStart`;
}

// Report an API success
export function apiSuccess( type ) {
  return `${type}${SUCCESS_SUFFIX}/apiSuccess`;
}

// Report an API error
export function apiError( type ) {
  return `${type}${STARTED_SUFFIX}/apiError`;
}

// Report a timeout start
export function timeoutStart( type ) {
  return `${type}${STARTED_SUFFIX}/timeoutStart`;
}

// Report a timeout end
export function timeoutEnd( type ) {
  return `${type}${STARTED_SUFFIX}/timeoutEnd`;
}

// Execute a state change, goes directly to the reducer with no side effects
export function execute( type ) {
  return `${type}${EXECUTED_SUFFIX}/execute`;
}
