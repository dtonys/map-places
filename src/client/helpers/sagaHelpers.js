import {
  put,
  call,
  take,
  fork,
} from 'redux-saga/effects';

import makeAction, {
  request,
  requestFork,
  apiStart,
  apiSuccess,
  apiError,
  localStorageStart,
  localStorageSuccess,
  localStorageError,
} from 'helpers/reduxAction';


// export function* apiFlow( api, actionName, deferred, isLocalStorage ) {
export function* apiFlow( api, actionName, { deferred, isLocalStorage }) {
  const start = isLocalStorage ? localStorageStart : apiStart;
  const success = isLocalStorage ? localStorageSuccess : apiSuccess;
  const error = isLocalStorage ? localStorageError : apiError;

  let responseData = null;
  const dispatchedActionsTracker = {
    log: [],
    startAction: null,
    successAction: null,
    errorAction: null,
    addAction: function ( trackerType, action ) { // eslint-disable-line func-names
      if ( ~Object.keys(this).indexOf(trackerType) ) {
        this[trackerType] = action;
      }
      this.log.push( action );
    },
  };

  const _startedAction = makeAction( start(actionName) );
  yield put( _startedAction );
  dispatchedActionsTracker.addAction('startAction', _startedAction);
  try {
    responseData = yield call( api );
    const _successAction = makeAction( success(actionName), responseData );
    yield put( _successAction );
    dispatchedActionsTracker.addAction('successAction', _successAction);
    if ( deferred ) {
      deferred.resolve(_successAction);
    }
  }
  catch ( errorData ) {
    const _errorAction = makeAction( error(actionName), errorData );
    yield put( _errorAction );
    responseData = errorData;
    dispatchedActionsTracker.addAction('errorAction', _errorAction);
    if ( deferred ) {
      deferred.reject(_errorAction);
    }
  }

  return dispatchedActionsTracker;
}

export function createSagaWatcher( actionType, actionRunner ) {
  return function* () { // eslint-disable-line func-names
    while ( true ) { // eslint-disable-line no-constant-condition
      const action =  yield take([
        actionType,
        request(actionType),
        requestFork(actionType),
      ]);
      if ( action.type === request(actionType) ) {
        yield* actionRunner( action );
      }
      if ( action.type === requestFork(actionType) ) {
        fork( actionRunner, action );
      }
    }
  };
}

