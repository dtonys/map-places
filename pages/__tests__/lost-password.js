jest.mock('superagent', () => ( require('../../jest-config/__mocks__/superagent') ));
import { getMockUrlProp } from './helpers/utils';
import { mount } from 'enzyme';
import LostPasswordPage from '../lost-password';


describe('Lost password page', () => {

  let superagent = null;
  let lostPasswordPage = null;
  beforeAll((done) => {
    superagent = require('superagent');
    superagent.__setMockDelay(1);
    lostPasswordPage = mount(<LostPasswordPage url={getMockUrlProp('/lost-password')} />);
    done();
  });

  test('Should mount and render successfully', (done) => {
    expect(lostPasswordPage).toMatchSnapshot();
    done();
  });

  test('On submit error, should show error message', (done) => {
    const errorResponse = {
      error: [
        { message: 'Email not found' },
      ],
    };
    superagent.__setMockResponse({
      status: 404,
      text: JSON.stringify(errorResponse),
    });
    lostPasswordPage.find('[data-test="email"]')
      .simulate('change', { target: { value: 'abcdefg@gmail.com' } });
    lostPasswordPage.find('[data-test="submit"]').getDOMNode().click();
    setTimeout(() => {
      expect(lostPasswordPage.find('form')).toIncludeText('Email not found');
      done();
    }, 100);
  });

  test('On submit success, should show a success screen', (done) => {
    const successResponse = {
      data: null,
    };
    superagent.__setMockResponse({
      status: 200,
      text: JSON.stringify(successResponse),
    });
    lostPasswordPage.find('[data-test="email"]')
      .simulate('change', { target: { value: 'abcdefg@gmail.com' } });
    lostPasswordPage.find('[data-test="submit"]').getDOMNode().click();
    setTimeout(() => {
      expect(lostPasswordPage.find('form'))
        .toIncludeText('Check your email to reset your password.');
      done();
    }, 100);
    done();
  });

});
