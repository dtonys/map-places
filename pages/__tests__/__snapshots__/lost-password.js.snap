// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lost password page Should mount and render successfully 1`] = `
<ClientEntryHOC(AttachReduxHOC(Connect(LostPasswordPage)))
  url={
    Object {
      "asPath": "/lost-password",
      "pathname": "/lost-password",
      "query": Object {},
    }
  }
>
  <AttachReduxHOC(Connect(LostPasswordPage))
    url={
      Object {
        "asPath": "/lost-password",
        "pathname": "/lost-password",
        "query": Object {},
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(LostPasswordPage)
        url={
          Object {
            "asPath": "/lost-password",
            "pathname": "/lost-password",
            "query": Object {},
          }
        }
      >
        <LostPasswordPage
          dispatch={[Function]}
          lostPasswordErrorMessage={null}
          url={
            Object {
              "asPath": "/lost-password",
              "pathname": "/lost-password",
              "query": Object {},
            }
          }
        >
          <div
            className="jsx-2986988504"
          >
            <Layout
              htmlTitle="Lost Password"
            >
              <div
                className="jsx-2650881538"
              >
                <SideEffect(Head)>
                  <Head />
                </SideEffect(Head)>
                <Connect(Navbar)>
                  <Navbar
                    currentUser={null}
                    dispatch={[Function]}
                    userAuthenticated={false}
                  >
                    <div
                      className="jsx-306046629 navbar"
                    >
                      <div
                        className="jsx-306046629 section section-1"
                      >
                        <LinkRoutes
                          route="/"
                        >
                          <Link
                            as="/"
                            href="/"
                          >
                            <a
                              className="jsx-306046629 logoLink"
                              href="/"
                              onClick={[Function]}
                            >
                              <div
                                className="jsx-306046629 logoName"
                              >
                                Map Places
                              </div>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <div
                          className="jsx-306046629 item"
                        >
                          Maps
                        </div>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <div
                          className="jsx-306046629 item"
                        >
                          Other
                        </div>
                      </div>
                      <div
                        className="jsx-306046629 section section-2"
                      />
                      <div
                        className="jsx-306046629 section section-3"
                      >
                        <LinkRoutes
                          route="/login"
                        >
                          <Link
                            as="/login"
                            href="/login"
                          >
                            <a
                              className="jsx-306046629"
                              href="/login"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline"
                              >
                                Login
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <LinkRoutes
                          route="/signup"
                        >
                          <Link
                            as="/signup"
                            href="/signup"
                          >
                            <a
                              className="jsx-306046629"
                              href="/signup"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline secondary"
                              >
                                Signup
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <LinkRoutes
                          route="/users/profile"
                        >
                          <Link
                            as="/users/profile"
                            href="/users?id=profile"
                          >
                            <a
                              className="jsx-306046629"
                              href="/users/profile"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline secondary"
                              >
                                Profile
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                      </div>
                      <JSXStyle
                        css=".navbar.jsx-306046629{height:60px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0 20px;background:#eee;position:absolute;top:0;left:0;right:0;border-bottom:solid black 2px;}.logoName.jsx-306046629{font-size:20px;color:rgb(49,52,57);cursor:pointer;text-decoration:none;}.logoLink.jsx-306046629{text-decoration:none;}.section-1.jsx-306046629{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-right:20px;white-space:nowrap;}.section-2.jsx-306046629{-webkit-flex:1;-ms-flex:1;flex:1;}.section-3.jsx-306046629{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;margin-left:20px;}.userName.jsx-306046629{margin-top:10px;overflow:hidden;text-overflow:ellipsis;max-width:200px;white-space:nowrap;}.searchWrap.jsx-306046629{margin:0 20px;}"
                        styleId="306046629"
                      />
                    </div>
                  </Navbar>
                </Connect(Navbar)>
                <div
                  className="jsx-2650881538 scrollWrap"
                >
                  <div
                    className="jsx-2650881538 content"
                  >
                    <div
                      className="jsx-2986988504"
                    >
                      <div
                        className="jsx-2986988504"
                        style={
                          Object {
                            "marginTop": "50px",
                          }
                        }
                      />
                      <ReduxForm
                        onSubmit={[Function]}
                        serverErrorMessage={null}
                      >
                        <Connect(Form(LostPasswordFormView))
                          destroyOnUnmount={true}
                          enableReinitialize={false}
                          forceUnregisterOnUnmount={false}
                          form="lost-password"
                          getFormState={[Function]}
                          keepDirtyOnReinitialize={false}
                          onSubmit={[Function]}
                          persistentSubmitErrors={false}
                          pure={true}
                          serverErrorMessage={null}
                          shouldAsyncValidate={[Function]}
                          shouldError={[Function]}
                          shouldValidate={[Function]}
                          shouldWarn={[Function]}
                          touchOnBlur={true}
                          touchOnChange={false}
                          validate={[Function]}
                        >
                          <Form(LostPasswordFormView)
                            anyTouched={false}
                            array={
                              Object {
                                "insert": [Function],
                                "move": [Function],
                                "pop": [Function],
                                "push": [Function],
                                "remove": [Function],
                                "removeAll": [Function],
                                "shift": [Function],
                                "splice": [Function],
                                "swap": [Function],
                                "unshift": [Function],
                              }
                            }
                            arrayInsert={[Function]}
                            arrayMove={[Function]}
                            arrayPop={[Function]}
                            arrayPush={[Function]}
                            arrayRemove={[Function]}
                            arrayRemoveAll={[Function]}
                            arrayShift={[Function]}
                            arraySplice={[Function]}
                            arraySwap={[Function]}
                            arrayUnshift={[Function]}
                            asyncValidating={false}
                            autofill={[Function]}
                            blur={[Function]}
                            change={[Function]}
                            clearAsyncError={[Function]}
                            clearSubmit={[Function]}
                            clearSubmitErrors={[Function]}
                            destroy={[Function]}
                            destroyOnUnmount={true}
                            dirty={false}
                            dispatch={[Function]}
                            enableReinitialize={false}
                            focus={[Function]}
                            forceUnregisterOnUnmount={false}
                            form="lost-password"
                            getFormState={[Function]}
                            initialize={[Function]}
                            initialized={false}
                            invalid={true}
                            keepDirtyOnReinitialize={false}
                            onSubmit={[Function]}
                            persistentSubmitErrors={false}
                            pristine={true}
                            pure={true}
                            registerField={[Function]}
                            registeredFields={
                              Object {
                                "email": Object {
                                  "count": 1,
                                  "name": "email",
                                  "type": "Field",
                                },
                              }
                            }
                            reset={[Function]}
                            serverErrorMessage={null}
                            setSubmitFailed={[Function]}
                            setSubmitSucceeded={[Function]}
                            shouldAsyncValidate={[Function]}
                            shouldError={[Function]}
                            shouldValidate={[Function]}
                            shouldWarn={[Function]}
                            startAsyncValidation={[Function]}
                            startSubmit={[Function]}
                            stopAsyncValidation={[Function]}
                            stopSubmit={[Function]}
                            submit={[Function]}
                            submitFailed={false}
                            submitSucceeded={false}
                            submitting={false}
                            syncErrors={
                              Object {
                                "email": Array [
                                  "Required field",
                                  "Invalid email",
                                ],
                              }
                            }
                            syncWarnings={Object {}}
                            touch={[Function]}
                            touchOnBlur={true}
                            touchOnChange={false}
                            unregisterField={[Function]}
                            untouch={[Function]}
                            updateSyncErrors={[Function]}
                            updateSyncWarnings={[Function]}
                            valid={false}
                            validExceptSubmit={false}
                            validate={[Function]}
                            values={Object {}}
                          >
                            <LostPasswordFormView
                              anyTouched={false}
                              array={
                                Object {
                                  "insert": [Function],
                                  "move": [Function],
                                  "pop": [Function],
                                  "push": [Function],
                                  "remove": [Function],
                                  "removeAll": [Function],
                                  "shift": [Function],
                                  "splice": [Function],
                                  "swap": [Function],
                                  "unshift": [Function],
                                }
                              }
                              asyncValidate={[Function]}
                              asyncValidating={false}
                              autofill={[Function]}
                              blur={[Function]}
                              change={[Function]}
                              clearAsyncError={[Function]}
                              clearSubmit={[Function]}
                              clearSubmitErrors={[Function]}
                              destroy={[Function]}
                              dirty={false}
                              dispatch={[Function]}
                              form="lost-password"
                              handleSubmit={[Function]}
                              initialize={[Function]}
                              initialized={false}
                              invalid={true}
                              onSubmit={[Function]}
                              pristine={true}
                              pure={true}
                              reset={[Function]}
                              serverErrorMessage={null}
                              submit={[Function]}
                              submitFailed={false}
                              submitSucceeded={false}
                              submitting={false}
                              touch={[Function]}
                              untouch={[Function]}
                              valid={false}
                              validate={[Function]}
                            >
                              <div
                                className="jsx-1368968365 formWrap"
                              >
                                <form
                                  action=""
                                  className="jsx-1368968365 form lostPasswordForm"
                                  method="post"
                                  onSubmit={[Function]}
                                >
                                  <h3
                                    className="jsx-1368968365"
                                  >
                                     Lost Password 
                                  </h3>
                                  <fieldset
                                    className="jsx-1368968365"
                                  >
                                    <Field
                                      component={[Function]}
                                      label="Email"
                                      name="email"
                                      requiredStar={true}
                                      type="email"
                                    >
                                      <Connect(ConnectedField)
                                        _reduxForm={
                                          Object {
                                            "anyTouched": false,
                                            "array": Object {
                                              "insert": [Function],
                                              "move": [Function],
                                              "pop": [Function],
                                              "push": [Function],
                                              "remove": [Function],
                                              "removeAll": [Function],
                                              "shift": [Function],
                                              "splice": [Function],
                                              "swap": [Function],
                                              "unshift": [Function],
                                            },
                                            "arrayInsert": [Function],
                                            "arrayMove": [Function],
                                            "arrayPop": [Function],
                                            "arrayPush": [Function],
                                            "arrayRemove": [Function],
                                            "arrayRemoveAll": [Function],
                                            "arrayShift": [Function],
                                            "arraySplice": [Function],
                                            "arraySwap": [Function],
                                            "arrayUnshift": [Function],
                                            "asyncErrors": undefined,
                                            "asyncValidate": [Function],
                                            "asyncValidating": false,
                                            "autofill": [Function],
                                            "blur": [Function],
                                            "change": [Function],
                                            "clearAsyncError": [Function],
                                            "clearSubmit": [Function],
                                            "clearSubmitErrors": [Function],
                                            "destroy": [Function],
                                            "destroyOnUnmount": true,
                                            "dirty": false,
                                            "dispatch": [Function],
                                            "enableReinitialize": false,
                                            "error": undefined,
                                            "focus": [Function],
                                            "forceUnregisterOnUnmount": false,
                                            "form": "lost-password",
                                            "getFormState": [Function],
                                            "getValues": [Function],
                                            "initialValues": undefined,
                                            "initialize": [Function],
                                            "initialized": false,
                                            "invalid": true,
                                            "keepDirtyOnReinitialize": false,
                                            "onSubmit": [Function],
                                            "persistentSubmitErrors": false,
                                            "pristine": true,
                                            "pure": true,
                                            "register": [Function],
                                            "registerField": [Function],
                                            "registerInnerOnSubmit": [Function],
                                            "registeredFields": Object {
                                              "email": Object {
                                                "count": 1,
                                                "name": "email",
                                                "type": "Field",
                                              },
                                            },
                                            "reset": [Function],
                                            "sectionPrefix": undefined,
                                            "serverErrorMessage": null,
                                            "setSubmitFailed": [Function],
                                            "setSubmitSucceeded": [Function],
                                            "shouldAsyncValidate": [Function],
                                            "shouldError": [Function],
                                            "shouldValidate": [Function],
                                            "shouldWarn": [Function],
                                            "startAsyncValidation": [Function],
                                            "startSubmit": [Function],
                                            "stopAsyncValidation": [Function],
                                            "stopSubmit": [Function],
                                            "submit": [Function],
                                            "submitFailed": false,
                                            "submitSucceeded": false,
                                            "submitting": false,
                                            "syncErrors": Object {
                                              "email": Array [
                                                "Required field",
                                                "Invalid email",
                                              ],
                                            },
                                            "syncWarnings": Object {},
                                            "touch": [Function],
                                            "touchOnBlur": true,
                                            "touchOnChange": false,
                                            "triggerSubmit": undefined,
                                            "unregister": [Function],
                                            "unregisterField": [Function],
                                            "untouch": [Function],
                                            "updateSyncErrors": [Function],
                                            "updateSyncWarnings": [Function],
                                            "valid": false,
                                            "validExceptSubmit": false,
                                            "validate": [Function],
                                            "values": Object {},
                                            "warning": undefined,
                                          }
                                        }
                                        component={[Function]}
                                        label="Email"
                                        name="email"
                                        normalize={[Function]}
                                        requiredStar={true}
                                        type="email"
                                      >
                                        <ConnectedField
                                          _reduxForm={
                                            Object {
                                              "anyTouched": false,
                                              "array": Object {
                                                "insert": [Function],
                                                "move": [Function],
                                                "pop": [Function],
                                                "push": [Function],
                                                "remove": [Function],
                                                "removeAll": [Function],
                                                "shift": [Function],
                                                "splice": [Function],
                                                "swap": [Function],
                                                "unshift": [Function],
                                              },
                                              "arrayInsert": [Function],
                                              "arrayMove": [Function],
                                              "arrayPop": [Function],
                                              "arrayPush": [Function],
                                              "arrayRemove": [Function],
                                              "arrayRemoveAll": [Function],
                                              "arrayShift": [Function],
                                              "arraySplice": [Function],
                                              "arraySwap": [Function],
                                              "arrayUnshift": [Function],
                                              "asyncErrors": undefined,
                                              "asyncValidate": [Function],
                                              "asyncValidating": false,
                                              "autofill": [Function],
                                              "blur": [Function],
                                              "change": [Function],
                                              "clearAsyncError": [Function],
                                              "clearSubmit": [Function],
                                              "clearSubmitErrors": [Function],
                                              "destroy": [Function],
                                              "destroyOnUnmount": true,
                                              "dirty": false,
                                              "dispatch": [Function],
                                              "enableReinitialize": false,
                                              "error": undefined,
                                              "focus": [Function],
                                              "forceUnregisterOnUnmount": false,
                                              "form": "lost-password",
                                              "getFormState": [Function],
                                              "getValues": [Function],
                                              "initialValues": undefined,
                                              "initialize": [Function],
                                              "initialized": false,
                                              "invalid": true,
                                              "keepDirtyOnReinitialize": false,
                                              "onSubmit": [Function],
                                              "persistentSubmitErrors": false,
                                              "pristine": true,
                                              "pure": true,
                                              "register": [Function],
                                              "registerField": [Function],
                                              "registerInnerOnSubmit": [Function],
                                              "registeredFields": Object {
                                                "email": Object {
                                                  "count": 1,
                                                  "name": "email",
                                                  "type": "Field",
                                                },
                                              },
                                              "reset": [Function],
                                              "sectionPrefix": undefined,
                                              "serverErrorMessage": null,
                                              "setSubmitFailed": [Function],
                                              "setSubmitSucceeded": [Function],
                                              "shouldAsyncValidate": [Function],
                                              "shouldError": [Function],
                                              "shouldValidate": [Function],
                                              "shouldWarn": [Function],
                                              "startAsyncValidation": [Function],
                                              "startSubmit": [Function],
                                              "stopAsyncValidation": [Function],
                                              "stopSubmit": [Function],
                                              "submit": [Function],
                                              "submitFailed": false,
                                              "submitSucceeded": false,
                                              "submitting": false,
                                              "syncErrors": Object {
                                                "email": Array [
                                                  "Required field",
                                                  "Invalid email",
                                                ],
                                              },
                                              "syncWarnings": Object {},
                                              "touch": [Function],
                                              "touchOnBlur": true,
                                              "touchOnChange": false,
                                              "triggerSubmit": undefined,
                                              "unregister": [Function],
                                              "unregisterField": [Function],
                                              "untouch": [Function],
                                              "updateSyncErrors": [Function],
                                              "updateSyncWarnings": [Function],
                                              "valid": false,
                                              "validExceptSubmit": false,
                                              "validate": [Function],
                                              "values": Object {},
                                              "warning": undefined,
                                            }
                                          }
                                          asyncValidating={false}
                                          component={[Function]}
                                          dirty={false}
                                          dispatch={[Function]}
                                          label="Email"
                                          name="email"
                                          normalize={[Function]}
                                          pristine={true}
                                          requiredStar={true}
                                          syncError={
                                            Array [
                                              "Required field",
                                              "Invalid email",
                                            ]
                                          }
                                          type="email"
                                        >
                                          <TextInput
                                            input={
                                              Object {
                                                "name": "email",
                                                "onBlur": [Function],
                                                "onChange": [Function],
                                                "onDragStart": [Function],
                                                "onDrop": [Function],
                                                "onFocus": [Function],
                                                "value": "",
                                              }
                                            }
                                            label="Email"
                                            meta={
                                              Object {
                                                "active": false,
                                                "asyncValidating": false,
                                                "autofilled": false,
                                                "dirty": false,
                                                "dispatch": [Function],
                                                "error": Array [
                                                  "Required field",
                                                  "Invalid email",
                                                ],
                                                "form": "lost-password",
                                                "initial": undefined,
                                                "invalid": true,
                                                "pristine": true,
                                                "submitFailed": false,
                                                "submitting": false,
                                                "touched": false,
                                                "valid": false,
                                                "visited": false,
                                                "warning": undefined,
                                              }
                                            }
                                            requiredStar={true}
                                            type="email"
                                          >
                                            <div
                                              className="form-item"
                                            >
                                              <label>
                                                Email
                                                <span
                                                  className="req"
                                                  style={
                                                    Object {
                                                      "marginLeft": "2px",
                                                    }
                                                  }
                                                >
                                                  *
                                                </span>
                                              </label>
                                              <input
                                                className=""
                                                data-test="email"
                                                name="email"
                                                onBlur={[Function]}
                                                onChange={[Function]}
                                                onDragStart={[Function]}
                                                onDrop={[Function]}
                                                onFocus={[Function]}
                                                type="email"
                                                value=""
                                              />
                                            </div>
                                          </TextInput>
                                        </ConnectedField>
                                      </Connect(ConnectedField)>
                                    </Field>
                                    <div
                                      className="jsx-1368968365 form-item"
                                    >
                                      <button
                                        className="jsx-1368968365 w50"
                                        data-test="submit"
                                      >
                                        Submit
                                      </button>
                                    </div>
                                  </fieldset>
                                </form>
                                <JSXStyle
                                  css=".lostPasswordForm.jsx-1368968365{width:600px;margin:auto;}.formWrap.jsx-1368968365{padding:0 10px;}"
                                  styleId="1368968365"
                                />
                              </div>
                            </LostPasswordFormView>
                          </Form(LostPasswordFormView)>
                        </Connect(Form(LostPasswordFormView))>
                      </ReduxForm>
                    </div>
                  </div>
                </div>
                <JSXStyle
                  css=".content.jsx-2650881538{max-width:840px;margin:auto;}.scrollWrap.jsx-2650881538{position:absolute;top:60px;left:0;right:0;bottom:0;overflow:auto;}h1.jsx-2650881538{text-align:center;}"
                  styleId="2650881538"
                />
              </div>
            </Layout>
            <JSXStyle
              css=".successWrap.jsx-2986988504{width:600px;margin:auto;}.messageWrap.jsx-2986988504{border:1px solid rgba(0,0,0,.1);text-align:center;padding:2rem;}"
              styleId="2986988504"
            />
          </div>
        </LostPasswordPage>
      </Connect(LostPasswordPage)>
    </Provider>
  </AttachReduxHOC(Connect(LostPasswordPage))>
</ClientEntryHOC(AttachReduxHOC(Connect(LostPasswordPage)))>
`;
